lib/demos/reference/motion_demo_shared_y_axis_transition.dart

```diff
@@ -0,0 +1,167 @@
+import 'dart:math';
+import 'package:flutter/material.dart';
+import 'package:animations/animations.dart';
+import 'package:gallery/l10n/gallery_localizations.dart';
+
+// BEGIN sharedYAxisTransitionDemo
+
+class SharedYAxisTransitionDemo extends StatefulWidget {
+  const SharedYAxisTransitionDemo({Key key}) : super(key: key);
+
+  @override
+  _SharedYAxisTransitionDemoState createState() =>
+      _SharedYAxisTransitionDemoState();
+}
+
+class _SharedYAxisTransitionDemoState extends State<SharedYAxisTransitionDemo>
+    with SingleTickerProviderStateMixin {
+  bool _isAlphabetical = false;
+  AnimationController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = AnimationController(
+      duration: const Duration(milliseconds: 300),
+      vsync: this,
+    );
+  }
+
+  final _recentList = ListView(
+    // Adding [UniqueKey] to make sure the widget rebuilds when transitioning.
+    key: UniqueKey(),
+    children: [
+      ...List.generate(10, (index) => _AlbumTile((index + 1).toString()))
+    ],
+  );
+
+  final _alphabeticalList = ListView(
+    // Adding [UniqueKey] to make sure the widget rebuilds when transitioning.
+    key: UniqueKey(),
+    children: _alphabet.map((letter) => _AlbumTile(letter)).toList(),
```

Comment:
this can also be a collection for loop:

[
  for (final letter in _alphabet)
    _AlbumTile(letter)
]

Reply:
will do